homeassistant/components/sharkiq/sharkiq.py

```diff
@@ -0,0 +1,283 @@
+        """Connect to dispatcher listening for entity data notifications."""
+        self.async_on_remove(
+            self.coordinator.async_add_listener(self.async_write_ha_state)
+        )
+
+    async def async_return_to_base(self, **kwargs):
+        """Have the device return to base."""
+        await self.sharkiq.async_set_operating_mode(OperatingModes.RETURN)
+        await self.coordinator.async_refresh()
+
+    async def async_pause(self):
+        """Pause the cleaning task."""
+        await self.sharkiq.async_set_operating_mode(OperatingModes.PAUSE)
+        await self.coordinator.async_refresh()
+
+    async def async_start(self):
+        """Start the device."""
+        await self.sharkiq.async_set_operating_mode(OperatingModes.START)
+        await self.coordinator.async_refresh()
+
+    async def async_stop(self, **kwargs):
+        """Stop the device."""
+        await self.sharkiq.async_set_operating_mode(OperatingModes.STOP)
+        await self.coordinator.async_refresh()
+
+    async def async_locate(self, **kwargs):
+        """Cause the device to generate a loud chirp."""
+        await self.sharkiq.async_find_device()
+
+    @property
+    def fan_speed(self) -> str:
+        """Return the current fan speed."""
+        fan_speed = None
+        speed_level = self.sharkiq.get_property_value(Properties.POWER_MODE)
+        for k, val in FAN_SPEEDS_MAP.items():
+            if val == speed_level:
+                fan_speed = k
+        return fan_speed
+
+    async def async_set_fan_speed(self, fan_speed: str, **kwargs):
+        """Set the fan speed."""
+        await self.sharkiq.async_set_property_value(
+            Properties.POWER_MODE, FAN_SPEEDS_MAP.get(fan_speed.capitalize())
+        )
+        await self.coordinator.async_refresh()
+
+    @property
+    def fan_speed_list(self):
+        """Get the list of available fan speed steps of the vacuum cleaner."""
+        return list(FAN_SPEEDS_MAP.keys())
+
+    # Various attributes we want to expose
+    @property
+    def recharge_resume(self) -> Optional[bool]:
+        """Recharge and resume mode active."""
+        return self.sharkiq.get_property_value(Properties.RECHARGE_RESUME)
+
+    @property
+    def rssi(self) -> Optional[int]:
+        """Get the WiFi RSSI."""
+        return self.sharkiq.get_property_value(Properties.RSSI)
+
+    @property
+    def low_light(self):
+        """Let us know if the robot is operating in low-light mode."""
+        return self.sharkiq.get_property_value(Properties.LOW_LIGHT_MISSION)
+
+    @property
+    def shark_state_attributes(self) -> Dict:
+        """Return a dictionary of device state attributes specific to sharkiq."""
+        data = {
+            ATTR_ERROR_CODE: self.error_code,
+            ATTR_ERROR_MSG: self.sharkiq.error_text,
+            ATTR_LOW_LIGHT: self.low_light,
+            ATTR_RECHARGE_RESUME: self.recharge_resume,
+            ATTR_RSSI: self.rssi,
```

Comment:
We shouldn't include rssi. If anything, it should be it's own sensor but even if we add that, it should be disabled by default. It is very spammy and people rarely look at it.